spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceWebFilter.java

```diff
@@ -255,6 +262,15 @@ private void addClassNameTag(Object handler, Span span) {
 		return ORDER;
 	}
 
+	static final class DecoratedServerHttpResponse extends ServerHttpResponseDecorator {
+
+		String method, httpRoute;
```

Comment:
if you move instantiating this to where handleSend is called, you can make these final and always assign a value